### PR TITLE
Add inner join, multiple extractor example

### PR DESCRIPTION
**Pull request by hamishdickson**

### CODE REVIEW

File: examples/src/main/scala/sqlest/examples/Extractor.scala [hunk 1]
```diff
@@ -21,6 +21,25 @@ object ExtractorExamples extends App with DatabaseExample {
...
+
+  // extractors can be used in the declaration of other extractors
+  lazy val smoothyExtractor = extract[Smoothy](
+    description = SmoothyTable.description,
+    fruits = fruitExtractor.asList
+  ).groupBy(SmoothyTable.id)
+
+  // inner joins can be used as follows
+  val smoothies =
+    select(smoothyExtractor.columns)  // Extractors maintain a list of columns they use
```
Comment (brendanator) [inline examples/src/main/scala/sqlest/examples/Extractor.scala]: It's now not necessary to specify `smoothyExtractor.columns`. When extract... is called it sets the fields to select on the select statement.

This line can just be replaced with `select`

Comment (hamishdickson) [inline examples/src/main/scala/sqlest/examples/Extractor.scala]: Ah yes, fair point - I've changed that now!

File: examples/src/main/scala/sqlest/examples/Extractor.scala [hunk 2]
```diff
@@ -21,6 +21,25 @@ object ExtractorExamples extends App with DatabaseExample {
...
+  // extractors can be used in the declaration of other extractors
+  lazy val smoothyExtractor = extract[Smoothy](
+    description = SmoothyTable.description,
+    fruits = fruitExtractor.asList
+  ).groupBy(SmoothyTable.id)
+
+  // inner joins can be used as follows
+  val smoothies =
+    select
+      .from(
```
Comment (brendanator) [inline examples/src/main/scala/sqlest/examples/Extractor.scala]: Sorry I missed this the first time.
You can now write the inner joins separately to the from clauses so it looks more like regular sql:
```scala
  .from(SmoothyTable)
  .innerJoin(IngredientsTable).on(SmoothyTable.id === IngredientsTable.smoothyId)
  .innerJoin(FruitTable).on(IngredientsTable.fruitId === FruitTable.id)
```

Comment (hamishdickson) [inline examples/src/main/scala/sqlest/examples/Extractor.scala]: Ah - also helpful. I'll update the README at the same time (I took this example from there)

**Comment by hamishdickson** [comment]
Sorry there are so many commits here, I committed some stuff as I was refactoring by mistake. Next time I will split the changes out into branches and do it properly

**Comment by frankwallis** [comment]
what you need is an 'interactive rebase with squashed commits'

**Pull request merged**